rotate-onboarding-keys/main.go

```diff
@@ -57,90 +56,87 @@ func main() {
 	// Instead, a safer approach is to delete both secrets and then recreate them simultaneously
 	// to ensure consistency and accuracy of all secrets. By this way it will be easier to diagnose the
 	// issues if one or two secrets do not exist instead of trying to understand if they match
-	err = clientset.CoreV1().
-		Secrets(operatorNamespace).
-		Delete(ctx, onboardingValidationPrivateKeySecretName, metav1.DeleteOptions{})
+	privateSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      onboardingValidationPrivateKeySecretName,
+			Namespace: operatorNamespace,
+		},
+	}
+	err = cl.Delete(ctx, privateSecret, client.GracePeriodSeconds(0))
```

Comment:
why do we need graceperiod option here? I remember using this only w/ Pods 🤔.

Reply:
I'm also wondering about this, especially given that it was not being used previously. If it's a known difference in behavior and you're just maintaining previous behaviors, that's perfectly fine.

Reply:
@leelavg  I have created a new Pull Request (PR) for replacing the native client with runtime. you are Right its only used for pods, I have removed the grace period for deletion of pods in this PR.  please review the new PR and we can discuss any further comments over there. @jarrpa  Just wanted to let you know that the deletion of the secret was also present in the previous PR. Thanks!